### PR TITLE
Fix issue where valueLabel doesn't show if value is 0

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Resolves an issue where using `valueLabel` on Select and `value={0}` results in the valueLabel not being displayed correctly.

#### What testing has been done on this PR?
Tested with the following storybook story and added a jest test.
```javascript
import React from 'react';

import { Box, Select } from 'grommet';

export const Test = () => {
  const [value, setValue] = React.useState(0);
  return (
    <Box fill align="center" justify="start" pad="large" gap="medium">
      <Select
        options={[0, 1, 2]}
        value={value}
        valueLabel={(val) => `Value: ${val}`}
        onChange={({ option }) => setValue(option)}
      />
    </Box>
  );
};

export default {
  title: 'Input/Select/Test',
};
```

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/6829

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Select/Select.js [hunk 1]
```diff
@@ -233,7 +233,7 @@ const Select = forwardRef(
     // element to show, trumps inputValue
     const selectValue = useMemo(() => {
       if (valueLabel instanceof Function) {
-        if (value) return valueLabel(value);
+        if (value !== '' && value !== null) return valueLabel(value);
```
Comment (MikeKingdom) [inline src/js/components/Select/Select.js:236]: There are other 'falsey' values like `undefined` or `false` that this handled before. Should we be calling valueLabel for `undefined` or `false`? Might be better to change this to:
```javascript
if (value || value === 0) return valueLabel(value)
```

Comment (taysea) [reply src/js/components/Select/Select.js]: Not sure if this is a use case, but is it possible that someone might have some options like `[true, false]` ?

Comment (jcfilben) [reply src/js/components/Select/Select.js]: ^ good point, we do support having `[true, false]` as options. Maybe we could do something like `if (value || value === 0 || value === false) return valueLabel(value);`? Not sure if there is a cleaner way to write this to get our desired behavior.

Comment (jcfilben) [reply src/js/components/Select/Select.js]: Cool, just pushed up that change

**Review by taysea — APPROVED** [review]

**Review by MikeKingdom — APPROVED** [review]

**Pull request merged**